### PR TITLE
Enable require() in event scripts

### PR DESCRIPTION
**Pull request by n321203**
This patch enables require() in event scripts.

Please note that if you require() and run an asynchronous function in an event script, the event script might finish and return its value without waiting for your async function to complete.

(PS. Never mind the commit messages...)

### CODE REVIEW

**Comment by NicolasRitouet** [comment]
Sorry I changed the title, I though you were finished :relaxed:

**Comment by n321203** [comment]
Just went out of battery, now back online :)

File: lib/script.js [hunk 1]
```diff
@@ -36,6 +36,9 @@ Script.prototype.run = function (ctx, domain, fn) {
   
   var scriptContext = {
     'this': {},
+    require: function(m){
+      return require(m);
+    },
```
Comment (ericfong) [inline lib/script.js]: I think we should consider relative modules, like require('../../hello.js')

Comment (ericfong) [inline lib/script.js]: PR #496 may already have this?

**Comment by n321203** [comment]
Yup, seems like #496 already implements this.

File: lib/script.js (hunk 1 above)
Comment (n321203) [inline lib/script.js]: Yup, seems like #496 already implements this.

**Comment by andreialecu** [comment]
As already noted in a line diff, #496 exposes `require`.

Also, `require` is also available as part of an extension here: https://github.com/mcantelon/dpd-event-extension

**Comment by n321203** [comment]
Closing this, already handled by #496.

**Comment by projectxmaker** [comment]
hi all,

I don't see this patch in master branch ? or I missed sthing ???

**Pull request closed without merging**